src/main/java/ch/njol/skript/lang/function/FunctionRegistry.java

```diff
@@ -354,13 +354,45 @@ public Retrieval<Signature<?>> getSignature(
 		@NotNull Class<?>... args
 	) {
 		if (namespace == null) {
-			return getSignature(GLOBAL_NAMESPACE, FunctionIdentifier.of(name, false, args));
+			return getSignature(GLOBAL_NAMESPACE, FunctionIdentifier.of(name, false, args), false);
 		}
 
 		Retrieval<Signature<?>> attempt = getSignature(new NamespaceIdentifier(namespace),
-			FunctionIdentifier.of(name, true, args));
+			FunctionIdentifier.of(name, true, args), false);
 		if (attempt.result() == RetrievalResult.NOT_REGISTERED) {
-			return getSignature(GLOBAL_NAMESPACE, FunctionIdentifier.of(name, false, args));
+			return getSignature(GLOBAL_NAMESPACE, FunctionIdentifier.of(name, false, args), false);
+		}
+		return attempt;
+	}
+
+	/**
+	 * Gets the signature for a function with the given name and arguments. If no local function is found,
+	 * checks for global functions. If {@code namespace} is null, only global signatures will be checked.
+	 * <p>
+	 * This function checks performs no argument conversions, and is only used for determining whether a
+	 * signature already exists with the exact specified arguments. In almost all cases, {@link #getSignature(String, String, Class[])}
+	 * should be used.
+	 * </p>
+	 *
+	 * @param namespace The namespace to get the function from.
+	 *                  Usually represents the path of the script this function is registered in.
+	 * @param name      The name of the function.
+	 * @param args      The types of the arguments of the function.
+	 * @return The signature for the function with the given name and argument types, or null if no such function exists.
+	 */
+	Retrieval<Signature<?>> getExactSignature(
+		@Nullable String namespace,
+		@NotNull String name,
+		@NotNull Class<?>... args
+	) {
+		if (namespace == null) {
+			return getSignature(GLOBAL_NAMESPACE, FunctionIdentifier.of(name, false, args), true);
+		}
+
+		Retrieval<Signature<?>> attempt = getSignature(new NamespaceIdentifier(namespace),
+			FunctionIdentifier.of(name, true, args), true);
+		if (attempt.result() == RetrievalResult.NOT_REGISTERED) {
+			return getSignature(GLOBAL_NAMESPACE, FunctionIdentifier.of(name, false, args), true);
 		}
 		return attempt;
```

Comment:
Something like this might be better to reduce code duplication
```suggestion
		Retrieval<Signature<?>> attempt = null;
		if (namespace != null) {
			attempt = getSignature(new NamespaceIdentifier(namespace),
					FunctionIdentifier.of(name, true, args), true);
		}
		if (attempt == null || attempt.result() == RetrievalResult.NOT_REGISTERED) {
			attempt = getSignature(GLOBAL_NAMESPACE, FunctionIdentifier.of(name, false, args), true);
		}
		return attempt;
```